### PR TITLE
Kuku

### PR DESCRIPTION
**Pull request by abaevbog**
Alright, that's what I got. 
Honestly, I could've tested it better but so far all seems to be working. 
I have the editing of menu as get request, I fixed Phillip's breaking signing in and out (never gets old), and I added the endpoint to get all dishes.  
Haven't squashed it to one commit yet in case I need to change something. 
Best,
Kuku

### CODE REVIEW

File: app/controllers/menu_controller.rb [hunk 1]
```diff
@@ -31,52 +37,58 @@ def find_dishes(meal_type,day)
...
-            else 
-                error = "Unknown action" 
-            end
-             
-        rescue => e
-            error = e
+        if action == "remove" 
+            Menu.remove_dish_by_day_type(dish,day,type, permanent)
+        elsif action == "add"
+            Menu.add_dish_to_cycle(day,type,dish, permanent)
```
Comment (philipkiely) [inline app/controllers/menu_controller.rb]: Why does this need the "permanent" argument?

File: app/controllers/menu_controller.rb [hunk 2]
```diff
@@ -31,52 +37,58 @@ def find_dishes(meal_type,day)
...
-            type = params[:type]
-            dish = Dish.get_dish(dish_name)
-            day = 7*(week) + day + 1
-            if action == nil || dish_name == nil
-                error = "Action or dish name is missing in request body"
-            end
+    begin
+        date = params[:date]
+        action = params[:act]
+        dish_name = params[:dish]
```
Comment (philipkiely) [inline app/controllers/menu_controller.rb]: How is this going to parse a dish name with spaces?

File: app/controllers/menu_controller.rb [hunk 3]
```diff
@@ -31,52 +37,58 @@ def find_dishes(meal_type,day)
...
+        if action == "remove" 
+            Menu.remove_dish_by_day_type(dish,day,type, permanent)
+        elsif action == "add"
+            Menu.add_dish_to_cycle(day,type,dish, permanent)
+        else 
+            raise "Unknown action" 
         end
+            
+    rescue => e
+        flash[:error] = e
```
Comment (philipkiely) [inline app/controllers/menu_controller.rb]: Will this raise and push to front end an error if I try dish "kuku" that does not exist in DB?

File: app/controllers/menu_controller.rb (hunk 1 above)
Comment (abaevbog) [reply app/controllers/menu_controller.rb]: Cause one can want to either add the dish to a given day permanently or only temporarily, meaning only for the next day. That's my attempt to accommodate that.

File: app/controllers/menu_controller.rb (hunk 2 above)
Comment (abaevbog) [reply app/controllers/menu_controller.rb]: rails handles that. I tried it with Rice Pilaf via ajax call, and as long as you encode it properly it works

File: app/controllers/menu_controller.rb (hunk 3 above)
Comment (abaevbog) [reply app/controllers/menu_controller.rb]: Yes. It should render the menu page again but with a flash error "Dish doesn't exist" that you can display

**Pull request merged**